packages/edit-widgets/src/blocks/legacy-widget/edit/widget-select-control.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import { pickBy, isEmpty, map, filter } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { SelectControl } from '@wordpress/components';
+
+export default function WidgetSelectControl( {
+	availableLegacyWidgets,
+	currentWidget,
+	onChange,
+	label = __( 'Select a legacy widget to display:' ),
+	emptyLabel = __( 'There are no widgets available.' ),
+} ) {
+	const usedReferenceWidgetNames = useSelect( ( select ) =>
+		select( 'core/edit-widgets' )
+			.getReferenceWidgetBlocks()
+			.map( ( { attributes } ) => attributes?.referenceWidgetName )
+	);
+	const choosableLegacyWidgets = useMemo( () => {
+		const visible = pickBy(
+			availableLegacyWidgets,
+			( { isHidden } ) => ! isHidden
+		);
+
+		// Filter out used reference widgets
+		return filter( visible, ( widget, referenceWidgetName ) => {
```

Comment:
Using filter here was converting the value from an object to an array, which caused an issue when mapping the options for the select control, as `key` is supposed to be the widget name instead of a number.

I've changed it to use the `pickBy` above. 

However, I don't think an object should be used for SelectControl values, as it'll give unpredictable results in terms of ordering.